Mage.Sets/src/mage/cards/d/DescendantsFury.java

```diff
@@ -0,0 +1,196 @@
+
+/**
+ * @author Susucr
+ */
+public final class DescendantsFury extends CardImpl {
+
+    public DescendantsFury(UUID ownerId, CardSetInfo setInfo) {
+        super(ownerId, setInfo, new CardType[]{CardType.ENCHANTMENT}, "{3}{R}");
+
+        // Whenever one or more creatures you control deal combat damage to a player, you may sacrifice one of them. If you do, reveal cards from the top of your library until you reveal a creature card that shares a creature type with the sacrificed creature. Put that card onto the battlefield and the rest on the bottom of your library in a random order.
+        Ability ability = new DealCombatDamageControlledTriggeredAbility(
+                Zone.BATTLEFIELD,
+                new DoIfCostPaid(
+                        new DescendantsFuryEffect(),
+                        new DescendantsFurySacrificeCost()
+                ),
+                true
+        );
+
+        ability.addWatcher(new DamageThisCombatWatcher());
+        this.addAbility(ability);
+    }
+
+    private DescendantsFury(final DescendantsFury card) {
+        super(card);
+    }
+
+    @Override
+    public DescendantsFury copy() {
+        return new DescendantsFury(this);
+    }
+}
+
+class DescendantsFurySacrificeCost extends CostImpl implements SacrificeCost {
+    DescendantsFurySacrificeCost() {
+        this.text = "sacrifice one of them";
+    }
+
+    private DescendantsFurySacrificeCost(final DescendantsFurySacrificeCost cost) {
+        super(cost);
+    }
+
+    @Override
+    public DescendantsFurySacrificeCost copy() {
+        return new DescendantsFurySacrificeCost(this);
+    }
+
+    @Override
+    public boolean pay(Ability ability, Game game, Ability source, UUID controllerId, boolean noMana, Cost costToPay) {
+        DamageThisCombatWatcher watcher = game.getState().getWatcher(DamageThisCombatWatcher.class);
+        if (watcher == null) {
+            return false;
+        }
+        TargetPointer targetPointer = source.getEffects().get(0).getTargetPointer();
+        if (targetPointer == null) {
+            return false;
+        }
+        Player damagedPlayer = game.getPlayer(targetPointer.getFirst(game, source));
+        Player controller = game.getPlayer(source.getControllerId());
+        if (controller == null || damagedPlayer == null) {
+            return false;
+        }
+
+        List<Permanent> possibleSacrifice =
+                watcher.getPermanents(controller.getId(), damagedPlayer.getId())
+                       .stream()
+                       .map(p -> p.getPermanent(game))
+                       .filter(p -> p != null && controller.canPaySacrificeCost(p, source, controllerId, game))
+                       .collect(Collectors.toList());
+
+        if (possibleSacrifice.isEmpty()) {
+            return false;
+        }
+
+        FilterControlledPermanent filter = new FilterControlledPermanent();
+        filter.add(new PermanentInListPredicate(possibleSacrifice));
```

Comment:
Is there a reason not to use `PermanentReferenceInCollectionPredicate` like Kaito?

```
 FilterPermanent filter = new FilterPermanent();
        filter.add(new PermanentReferenceInCollectionPredicate(
                watcher.getPermanents(controller.getId(),damagedPlayer.getId())));
```

Reply:
I think I need to call `controller.canPaySacrificeCost(p, source, controllerId, game)` on the permanents
to not allow selecting "Can't be sacrificed" permanents. For instance with [[Assault Suite]].

That is why I loop through the permanents to filter some of them.

Reply:
Well I think that is how those are supposed to be filtered out, not that sure looking at the code.

Reply:
I see. How hard would it be to make a predicate for can be sacrificed? That would probably be a more sustainable method (perhaps related: #9586). I'd like to avoid using that old PermanentInListPredicate that compares object identity in new code (guess it should be tagged deprecated) but I suppose it's okay here since the list is entirely internal to the effect.

Reply:
#9375 is the actual issue

Reply:
Alright, I'll see If I can manage to do that.
If not, I'll make a version with just PermanentReferenceInCollectionPredicate like Kaito, as I was expecting canPaySacrificeCost to help me there, but it does not seem to do so.

Reply:
Looks like `PlayerImpl::canPaySacrificeCost` only checks against Angel of Jubilation's effect right now.

Probably not worth the trouble until a separate major rework.

Reply:
Oh and even better... that's only for casting spells and activating abilities, and this is neither.

Reply:
ok, I'll clean up and this will bugged like most of the other sacrifice cost :(

Reply:
At least Yasharn is not an issue there.